monitoring/grafana/dashboards/tests/util.py

```diff
@@ -49,15 +50,21 @@ def add_dashboard_queries(data: Dict[str, Any], dashboard_data: Dict[str, Any],
             for target in panel['targets']:
                 title = panel['title']
                 legend_format = target['legendFormat'] if 'legendFormat' in target else ""
-                query_id = title + '-' + legend_format
+                query_id = f'{title}-{legend_format}'
                 if query_id in data['queries']:
                     # NOTE: If two or more panels have the same name and legend it
                     # might suggest a refactoring is needed or add something else
                     # to identify each query.
-                    cprint((f'WARNING: Query in panel "{title}" with legend "{legend_format}"'
-                                       ' already exists'), 'yellow')
+                    conflict_file = data['queries'][query_id]['path'].split('/')[-1]
```

Comment:
There's a `.rsplit('/', 1)` for these cases. But, given you're working with paths, using split with a literal "/" is not the portable way of processing paths. You shouldn't worry about / or \, since that's an implementation detail of the OS.

One possibility is using a high-level library for file management, like [pathlib](https://docs.python.org/3/library/pathlib.html). Instead of storing strings to paths, you can store `Path` objects, and then you do a `path.basename()` to get the filename.